osu.Game/Screens/Select/Footer.cs

```diff
@@ -60,10 +65,24 @@ private void showOverlay(OverlayContainer overlay)
 
         private void updateModeLight() => modeLight.FadeColour(buttons.FirstOrDefault(b => b.IsHovered)?.SelectedColour ?? Color4.Transparent, TRANSITION_LENGTH, Easing.OutQuint);
 
+        // Extra  pacing between FooterButtons is added at all UI scales.
+        // The higher the UI scale, the less spacing added.
+        // Maximum extra spacing  is 0.625f at 1.6 UI.
+        // Minimum extra spacing  is 0.320f at 0.8 UI.
+        private void updateSpacing(ValueChangedEvent<float> args) => buttons.Spacing = new Vector2(-FooterButton.SHEAR_WIDTH + (1f / args.NewValue) * 0.5f, 0);
```

Comment:
this still needs proper investigation. i'd hope to avoid ui spacing specific adjustments if possible.

Reply:
The UI scaling based approach was meant to counteract this remaining overlap. 

![Screenshot_20200222_214246](https://user-images.githubusercontent.com/43511642/75094173-cc223900-55c3-11ea-982e-6b1ab11e00a7.png)

Without the UI scaling elements, the overlap is still visible but not as obvious as before. You probably would miss it if I didnt mention that the overlap is still there. Still, it is kind of a half-hearted approach. If this is acceptable, then I would push the remaning commits and fix the conflicts. Else, I would be re-closing this PR.

Theoretically, the problems should already be solved when ``Masking=true`` is set, as it should mask the pixels outside of respective containers already. I tried investigating, but still can't determine why the pixels are "leaking out".

Reply:
Does the overlap depicted above grow & shrink with size with UI scaling changes? It seems to be a pixel-ish wide in size from the attached screenshot, which makes me think it is due to blending & antialiasing at the drawable edges causing a single pixel of overlap. If it does grow then it probably is something else.

Reply:
It looks like it stays at the same size, but it gets more obvious? (brighter?) when UI scale is smaller.

All images below taken at 1080p

UI Scale at 1.6
![image](https://user-images.githubusercontent.com/43511642/75094552-2244ab80-55c7-11ea-8950-4c2ca0b86d08.png)

UI Scale at 1.0 
![image](https://user-images.githubusercontent.com/43511642/75094565-486a4b80-55c7-11ea-877b-6f6af765d0ae.png)

UI Scale at 0.8
![image](https://user-images.githubusercontent.com/43511642/75094579-72237280-55c7-11ea-8d04-2be58a5b0ce1.png)